core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/iteration/IterateListSimultaneousltUnitTest.java

```diff
@@ -0,0 +1,74 @@
+package com.baeldung.list.iteration;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.data.util.StreamUtils;
+
+public class IterateListSimultaneousltUnitTest {
+
+    List<String> names = List.of("John", "Nik", "Tom");
+    List<Integer> numbers = List.of(1, 2, 3);
+
+    @Test
+    public void givenTwoLists_whenProcessByZipping_thenProcessedSimultaneously() {
+        Stream<String> nameStream = names.stream();
+        Stream<Integer> numberStream = numbers.stream();
+        List<String> processedList = StreamUtils.zip(numberStream, nameStream,
+            (numStr, nameStr) -> String.format("[Using Zip] Processing ArrayList Simultaneously : %d, %s", numStr, nameStr))
+            .collect(Collectors.toList());
+        Assertions.assertTrue(processedList.containsAll(
+          List.of("[Using Zip] Processing ArrayList Simultaneously : 1, John",
```

Comment:
Let's keep the text here a lot simpler so that the reader can tell what's going on. The word simultaneously is misleading. Just `1. John`, `2. Nik` would be enough to prove the point.